WT-6882 Files created during incremental backup should be copied in full.

sueloverso opened the pull request:
@keitharnoldsmith the change turned out to be a lot smaller than expected. I will open a separate ticket to modify the test_incr_backup to vary the allocation size. There have been a handful (7-8) failures noted in the ticket. I have run this change on the seeds from all failures successfully with this change.

keitharnoldsmith reviewed (approved):
Nice!  Glad it turned out to be simpler than expected.

LGTM!

sueloverso commented:
@keitharnoldsmith I reverted the original change and took a slightly different approach to handle across a reopen. @ddanderson, you can read the analysis in the ticket about what is root cause. I added variance for allocation size and closing/reopening the connection to `incr_backup`. I reran all 6 failing seeds on this branch and they pass.

sueloverso commented:
I have also kicked off a script to run randomized `test_incr_backup` in a loop and will let that run for several hours.

ddanderson commented on src/include/os.h:
```diff
@@ -111,6 +111,7 @@ struct __wt_fh {
     TAILQ_ENTRY(__wt_fh) hashq;     /* internal hash queue */
     u_int ref;                      /* reference count */
     WT_FS_OPEN_FILE_TYPE file_type; /* file type */
+    bool created_during_bkup;       /* file created during incremental backup */
```
Looks like this field is unused.

sueloverso replied:
Good catch. That was from an earlier iteration/attempt. Fixed.

ddanderson reviewed (approved):
LGTM - one small cleanup.

keitharnoldsmith reviewed (approved):
LGTM!
